apm-sniffer/apm-sdk-plugin/rabbitmq-plugin/src/main/java/org/apache/skywalking/apm/plugin/rabbitmq/ChannelNConstructorInterceptor.java

```diff
@@ -21,12 +21,24 @@
 import com.rabbitmq.client.Connection;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.util.StringUtil;
 
 public class ChannelNConstructorInterceptor implements InstanceConstructorInterceptor {
     @Override
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
         Connection connection = (Connection) allArguments[0];
-        String url = connection.getAddress().toString().replace("/", "") + ":" + connection.getPort();
+
+        String[] parts = connection.getAddress().toString().split("/");
+        String address;
+        if (StringUtil.isNotEmpty(parts[0])) {
+            address = parts[0];
+        } else if (parts.length >= 2 && StringUtil.isNotEmpty(parts[1])) {
+            address = parts[1];
+        } else {
+            address = "";
```

Comment:
If you can't resolve the address, this should not be an empty address. 
`connection.getAddress().toString()` could be used.